src/components/autocomplete/js/autocompleteDirective.js

```diff
@@ -397,7 +403,48 @@ function MdAutocomplete ($$mdSvgRegistry) {
             ? '<li ng-if="$mdAutocompleteCtrl.notFoundVisible()"\
                          md-autocomplete-parent-scope>' + template + '</li>'
             : '';
+      }
+
+      function getContainer(menuContainerClass, repeatMode) {
+        // prepend a space if needed
+        menuContainerClass = menuContainerClass ? ' ' + menuContainerClass : '';
+
+        if (isVirtualRepeatDisabled(repeatMode)) {
+          return '\
+            <div \
+                ng-hide="$mdAutocompleteCtrl.hidden"\
+                class="md-standard-list-container md-autocomplete-suggestions-container md-whiteframe-z1' + menuContainerClass + '"\
+                ng-class="{ \'md-not-found\': $mdAutocompleteCtrl.notFoundVisible() }"\
+                role="presentation">\
+              <div class="md-autocomplete-list-scroller" role="presentation">';
+        }
+
+        return '\
+          <md-virtual-repeat-container\
+              md-auto-shrink\
+              md-auto-shrink-min="1"\
+              ng-hide="$mdAutocompleteCtrl.hidden"\
+              class="md-virtual-repeat-container md-autocomplete-suggestions-container md-whiteframe-z1' + menuContainerClass + '"\
+              ng-class="{ \'md-not-found\': $mdAutocompleteCtrl.notFoundVisible() }"\
+              role="presentation">';
+      }
+
+      function getContainerClosingTags(repeatMode) {
+        return isVirtualRepeatDisabled(repeatMode) ?
+            '   </div>\
+              </div>\
+            </div>' : '</md-virtual-repeat-container>';
+      }
+
+      function getRepeatType(repeatMode) {
+        return isVirtualRepeatDisabled(repeatMode)  ?
+          'ng-repeat' : 'md-virtual-repeat';
+      }
 
+      function isVirtualRepeatDisabled(repeatMode) {
+        // ensure we have a valid repeat mode
+        repeatMode = getRepeatMode(repeatMode);
```

Comment:
I would prefer that this value is put in a separate variable rather than possibly modifying the parameter that was passed in.

Reply:
:+1: